librz/core/disasm.c

```diff
@@ -5274,9 +5319,37 @@ static void ds_end_line_highlight(RzDisasmState *ds) {
 	}
 }
 
-// int l is for lines
-RZ_API int rz_core_print_disasm(RzPrint *p, RzCore *core, ut64 addr, ut8 *buf, int len, int l, int invbreak, int cbytes, bool json, PJ *pj, RzAnalysisFunction *pdf) {
-	int continueoninvbreak = (len == l) && invbreak;
+/**
+ * \brief Free RzAnalysisDisasmText \p p
+ */
+RZ_API void rz_analysis_disasm_text_free(RzAnalysisDisasmText *t) {
+	if (!t) {
+		return;
+	}
+	free(t->text);
+	free(t);
+}
+
+/**
+ * \brief Disassemble \p len bytes and \p nlines opcodes, or a funitcon,
+ * 	  restricted by \p len and \p nlines at the same time
+ * \param core RzCore reference
+ * \param addr Address
+ * \param buf Buffer
+ * \param len Bytes number
+ * \param nlines Opcode number
+ * \param options Disassemble Options
+ * \return Disassemble bytes number
+ */
+RZ_API int rz_core_print_disasm(RZ_NONNULL RzCore *core, ut64 addr, RZ_NONNULL ut8 *buf, int len, int nlines,
+	RZ_NULLABLE RzCoreDisasmOptions *options) {
+	rz_return_val_if_fail(core && buf, 0);
+
+	PJ *pj = options ? options->pj : NULL;
+	bool json = options ? options->json : false;
+
```

Comment:
Did you consider using `RzCmdStateOutput`?

Reply:
Also, pass that as argument to this function instead of part of `RzCoreDisasmOptions`